pkg/agent/clientset.go

```diff
@@ -218,15 +221,45 @@ func (cs *ClientSet) sync() {
 }
 
 func (cs *ClientSet) ServerCount() int {
+	fromResponses := "KNP server response headers"
+	fromLeases := "KNP lease count"
+	fromFallback := "fallback to 1"
+
 	var serverCount int
-	if cs.leaseCounter != nil {
-		serverCount = cs.leaseCounter.Count()
-	} else {
-		serverCount = cs.lastReceivedServerCount
+	var countSourceLabel string
+
+	switch cs.serverCountSource {
+	case "":
```

Comment:
Golang does not do fall through (https://go.dev/ref/spec#Expression_switches "the first one that equals the switch expression triggers execution of the statements of the associated case; the other cases are skipped") So I believe empty string is going to silently do nothing, which I don't think is what we want. 

Did you mean the following?
case "", "default":

Reply:
Thanks! Fixed.